stable/database/files/backup_hooks.py

```diff
@@ -153,22 +182,40 @@ def freeze_archive(unfreeze=False):
         subprocess.check_output(
             ["fsfreeze", action, ARCHIVE_DIR], stderr=subprocess.STDOUT
         )
-
+    else:
+        raise RuntimeError("not supported quiesce mode '{}'".format(FREEZE_MODE))
 
 def pre_backup(backup_id, payload):
     # Check that backup ID was specified
     if not backup_id:
         raise UserError("Backup ID not specified")
 
+    # Get nuodb processes
+    processes = get_nuodb_process_info()
+    if not processes:
+        raise RuntimeError("No nuodb process found")
+
     # Make sure we do not attempt to execute pre-backup hook if a backup is
     # already in progress, which may block indefinitely if writes to the
     # archive and journal directories are frozen
     if os.path.exists(ARCHIVE_BACKUP_ID_FILE):
-        raise UserError(
-            "Backup ID file {} exists. Execute post-backup hook to complete current backup.".format(
-                ARCHIVE_BACKUP_ID_FILE
+        # Check if the SM process was restarted which will invalidate the
+        # previous pre-backup operation using HotSnap.
+        stored_start_id = get_start_id()
+        if FREEZE_MODE == "hotsnap" and stored_start_id and processes[0]["sid"] != stored_start_id:
```

Comment:
What is special about freeze mode `hotsnap` when it comes to handling the presence of the backup file?

Reply:
I have an E2E test for that [here](https://github.com/nuodb/nuodb-helm-charts/pull/365/files#diff-2a9c2e48a29de00e39dd2e1784e4ababaaec6e085b7f0c91a79a93c10a009966R338-R370) but it doesn't fully exercise your scenario since it doesn't enable _restore from snapshot_.

I think in your scenario we just log. See https://github.com/nuodb/nuodb-helm-charts/blob/master/stable/database/files/nuosm#L460.

Reply:
But why are we cleaning up the backup files and trying the backup for `hotsnap` while erroring out for other backup modes?

Reply:
SM restart invalidates the backup ID when using `hotsnap`, which is not the case when using `fsfreeze`.

Reply:
So basically, we want to avoid deleting the backup files if freeze mode is `fsfreeze`, since that would block things completely. I think we can safely apply this logic for the `suspend` freeze mode.

Reply:
Right. I can make the following change:

```diff
@@ -202,7 +206,7 @@ def pre_backup(backup_id, payload):
         # Check if the SM process was restarted which will invalidate the
         # previous pre-backup operation using HotSnap.
         stored_start_id = get_start_id()
-        if FREEZE_MODE == "hotsnap" and stored_start_id and processes[0]["sid"] != stored_start_id:
+        if FREEZE_MODE != MODE_FSFREEZE and stored_start_id and processes[0]["sid"] != stored_start_id:
             # Remote the backup files from the previous pre-backup operation
             LOGGER.warning("Unexpected start ID: current=%s, stored=%s. " +
                             "SM process restarted while executing backup ID %s",
```